cloudfoundry/structures_app.go

```diff
@@ -12,12 +12,12 @@ import (
 )
 
 func ResourceDataToAppDeploy(d *schema.ResourceData) (appdeployers.AppDeploy, error) {
-	enableSSH := d.Get("enable_ssh").(bool)
+	enableSSH := BoolToNullBool(d.Get("enable_ssh").(bool))
 	if d.IsNewResource() {
 		// if user does not explicitly set allow_ssh
 		// it set allow ssh to true only during creation
-		if _, ok := d.GetOk("enable_ssh"); !ok {
-			enableSSH = true
+		if !enableSSH.IsSet {
```

Comment:
Because of the line above `enableSSH := BoolToNullBool(d.Get("enable_ssh").(bool))` you can never have false at this point, see definition of [BoolToNullBool](https://github.com/cloudfoundry-community/terraform-provider-cloudfoundry/blob/8d135b82b1fdadb42c8197b433c6d002f723f399/cloudfoundry/utils_clients.go#L18-L23)